### ISSUE TITLE
Feature request: reattach drives that went offline

### ISSUE THREAD

**Issue by asyncth** [koverstreet/bcachefs#449]
In my limited testing, if a drive goes offline and then comes back with a different name (say `/dev/sda1` goes offline for 10 seconds, then comes back as `/dev/sdb1`), bcachefs is unable to determine that those are the same devices and keeps trying to write to `sda1`. It would be very nice if bcachefs could handle that without remounting.

**Comment by jpsollie** [comment]
I think this would require udev to understand the bcachefs UUID, something which will be considered when bcachefs is getting mainlined, then we could ask udev to send a dbus message to the mount handler.  For now, I do not see any options here, as this would require a separate bcachefs handler to listen for kernel events

**Comment by asyncth** [comment]
Yeah, this is more of a feature that would be nice to see long-term, rather than something that needs to be done now.

**Comment by Komzpa** [comment]
Current source has a manual no-remount path for part of this now, but I think the automatic hotplug/renamed-path part of this issue is still open.

What I found on current `bcachefs-tools` `origin/master` (`2eba266601c38da437d5a5641a00ed89119395fd`):

- `bcachefs device online <device path>` exists and is described as "Re-add a device to a running filesystem" / "Bring a device online".
- The implementation reads the superblock from the path passed at online time, checks that the device is still a member of the running filesystem, attaches it, updates the member state, and queues a reconcile pending scan.
- `ktest` already has running-mount offline/online coverage in `tests/fs/bcachefs/replication.ktest` (`test_device_set_state_offline()` and `test_device_readd()`).

So if `/dev/sda1` disappears and the same member later reappears as `/dev/sdb1`, the current manual recovery path should be:

```sh
bcachefs device online /dev/sdb1
```

That does not appear to be the same as automatic reattach from udev/kernel events, though. I did not find an open PR adding a mount handler / udev integration that would notice the new path and online the member without an admin command, so I'd keep this issue open if the desired end state is automatic reattachment.
